app/views/locations/index.html.erb

```diff
@@ -11,57 +11,102 @@
+        <% end %>
+        <%= table.head do |head| %>
+          <%= head.row do |row| %>
+            <% row.cell(header: true, text: "Postcode", html_attributes: {
+              scope: "col",
+            }) %>
+            <% row.cell(header: true, text: "Location code", html_attributes: {
+              scope: "col",
+            }) %>
+            <% row.cell(header: true, text: "Status", html_attributes: {
+              scope: "col",
+            }) %>
+          <% end %>
         <% end %>
+        <% @locations.each do |location| %>
+          <%= table.body do |body| %>
+            <%= body.row do |row| %>
+              <% row.cell(text: simple_format(location_cell_postcode(location, scheme_location_path(@scheme, location)), { class: "govuk-!-font-weight-bold" }, wrapper_tag: "div")) %>
+              <% row.cell(text: location.id) %>
+              <% row.cell(text: status_tag(location.status)) %>
+              <% end %>
+          <% end %>
+        <% end %>
+      <% end %>
+      <%= govuk_button_link_to "Add a location", new_scheme_location_path(@scheme), secondary: true %>
+    </div>
+  </div>
+
+<% else %>
+  <%= govuk_table do |table| %>
+    <%= table.caption(classes: %w[govuk-!-font-size-19 govuk-!-font-weight-regular]) do |caption| %>
+      <%= render(SearchResultCaptionComponent.new(searched: @searched, count: @pagy.count, item_label:, total_count: @total_count, item: "locations", path: request.path)) %>
+    <% end %>
+    <%= table.head do |head| %>
+      <%= head.row do |row| %>
+        <% row.cell(header: true, text: "Code", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Postcode", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Units", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Common unit type", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Mobility type", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Local authority", html_attributes: {
+          scope: "col",
+        }) %>
+        <% row.cell(header: true, text: "Available from", html_attributes: {
+          scope: "col",
+        }) %>
+      <% end %>
+    <% end %>
+    <% @locations.each do |location| %>
+      <%= table.body do |body| %>
+        <%= body.row do |row| %>
+          <% row.cell(text: location.id) %>
+          <% row.cell(text: simple_format(location_cell_postcode(location, scheme_location_path(@scheme, location)), { class: "govuk-!-font-weight-bold" }, wrapper_tag: "div")) %>
+          <% row.cell(text: location.units) %>
+          <% row.cell(text: simple_format("<span>#{location.type_of_unit}</span>")) %>
```

Comment:
```suggestion
          <% row.cell do %>
            <span><%= simple_format(location.type_of_unit %></span>
          <% end %>
```

I _think_ you can do this. When you are rendering HTML you should try and write it outside of Ruby, if you can.